### ISSUE TITLE
acme: sys: bus error on OpenBSD 6.6-current

### ISSUE THREAD

**Issue by mk-f** [9fans/plan9port#353]
acme dies with SIGBUS on OpenBSD 6.6-current  83b1e7a39b1e5af82846c7e9632b802c39d96f83:
```
$ ./o.acme                                                                        
acme: sys: bus error
9pserve <nil>: Jan 17 23:22:44.975 note: sys: bus error
```
```
Starting program: /home/user/src/plan9port/src/cmd/acme/o.acme 
[New process 64325]
[New process 90800]
[New thread 425779]
[New thread 350690]
[New thread 287617]
[New thread 177574]
[New thread 182322]

Thread 2.1 received signal SIGBUS, Bus error.
[Switching to thread 552737]
0x00000ae97348d937 in fmtstrinit (f=<optimized out>) at fmt/vsmprint.c:50
50		f->runes = 0;
#0  0x00000ae97348d937 in fmtstrinit (f=<optimized out>) at fmt/vsmprint.c:50
#1  vsmprint (fmt=0xae97342db8e "/tmp/ns.%s.%s", args=0xaeb82b13e88) at fmt/vsmprint.c:73
#2  0x00000ae97348c7df in smprint (fmt=0xae97342db8e "/tmp/ns.%s.%s") at fmt/smprint.c:14
#3  0x00000ae97348260e in nsfromdisplay () at getns.c:58
#4  getns () at getns.c:91
#5  0x00000ae973474bce in nsinit (name=0xae97342cb8d "plumb") at ns.c:13
#6  0x00000ae973474cae in nsmount (name=0xae97342db8e "/tmp/ns.%s.%s", aname=0xae97342d32f "") at ns.c:44
#7  0x00000ae973463f3b in plumbopenfid (name=0xae97342c7d7 "edit", mode=32) at fid.c:66
#8  0x00000ae97344fa5d in plumbthread (v=<optimized out>) at look.c:42
#9  0x00000ae97347acdf in threadstart (y=<optimized out>, x=<optimized out>) at thread.c:97
#10 0x0000000000000000 in ?? ()
Continuing.

Thread 2.3 received signal SIGBUS, Bus error.
[Switching to thread 350690]
_thread_sys_read () at -:3
3	-: No such file or directory.
#0  _thread_sys_read () at -:3
#1  0x00000aec619548ce in _libc_read_cancel (fd=3, buf=0xaeb85e880b0, nbytes=4) at /usr/src/lib/libc/sys/w_read.c:27
#2  0x00000ae973484252 in readn (f=3, av=0xaeb85e880b0, n=4) at readn.c:12
#3  0x00000ae97346bcaf in _drawrecv (mux=<optimized out>, canblock=<optimized out>, vp=0xaeb85e88170) at drawclient.c:177
#4  0x00000ae97346b370 in drawrecv (mux=0x3) at drawclient.c:198
#5  0x00000ae97347629a in _muxrecv (mux=0xaec2dd0a800, canblock=1, vp=0xaeb85e881e0) at io.c:98
#6  0x00000ae9734758d8 in muxrpc (mux=0xaec2dd0a800, tx=<optimized out>) at mux.c:141
#7  0x00000ae97346b514 in displayrpc (d=0xaebf1074800, tx=0xaeb85e88270, rx=0xaeb85e88438, freep=0x0) at drawclient.c:271
#8  0x00000ae97346b6a3 in _displayrdmouse (d=0x3, m=0xaeb85e88630, resized=0xaeb85e88654) at drawclient.c:323
#9  0x00000ae97346ed5b in _ioproc (arg=0xaec4b0eed00) at mouse.c:55
#10 0x00000ae973479237 in procmain (p=0xaebd9926800) at thread.c:338
#11 0x00000ae973476953 in startprocfn (v=<optimized out>) at pthread.c:97
#12 0x00000aebdf246531 in _rthread_start (v=<optimized out>) at /usr/src/lib/librthread/rthread.c:96
#13 0x00000aec6192d7e8 in __tfork_thread () at /usr/src/lib/libc/arch/amd64/sys/tfork_thread.S:77
#14 0x0000000000000000 in ?? ()
Continuing.

Thread 2.3 received signal SIGBUS, Bus error.
_thread_sys_read () at -:3
3	in -
#0  _thread_sys_read () at -:3
#1  0x00000aec619548ce in _libc_read_cancel (fd=3, buf=0xaeb85e880b0, nbytes=4) at /usr/src/lib/libc/sys/w_read.c:27
#2  0x00000ae973484252 in readn (f=3, av=0xaeb85e880b0, n=4) at readn.c:12
#3  0x00000ae97346bcaf in _drawrecv (mux=<optimized out>, canblock=<optimized out>, vp=0xaeb85e88170) at drawclient.c:177
#4  0x00000ae97346b370 in drawrecv (mux=0x3) at drawclient.c:198
#5  0x00000ae97347629a in _muxrecv (mux=0xaec2dd0a800, canblock=1, vp=0xaeb85e881e0) at io.c:98
#6  0x00000ae9734758d8 in muxrpc (mux=0xaec2dd0a800, tx=<optimized out>) at mux.c:141
#7  0x00000ae97346b514 in displayrpc (d=0xaebf1074800, tx=0xaeb85e88270, rx=0xaeb85e88438, freep=0x0) at drawclient.c:271
#8  0x00000ae97346b6a3 in _displayrdmouse (d=0x3, m=0xaeb85e88630, resized=0xaeb85e88654) at drawclient.c:323
#9  0x00000ae97346ed5b in _ioproc (arg=0xaec4b0eed00) at mouse.c:55
#10 0x00000ae973479237 in procmain (p=0xaebd9926800) at thread.c:338
#11 0x00000ae973476953 in startprocfn (v=<optimized out>) at pthread.c:97
#12 0x00000aebdf246531 in _rthread_start (v=<optimized out>) at /usr/src/lib/librthread/rthread.c:96
#13 0x00000aec6192d7e8 in __tfork_thread () at /usr/src/lib/libc/arch/amd64/sys/tfork_thread.S:77
#14 0x0000000000000000 in ?? ()
Continuing.
[Inferior 2 (process 90800) exited with code 01]
```
if you need any more info let me know.

**Comment by mk-f** [comment]
Seems to be related to https://github.com/9fans/plan9port/pull/350, which fixed the issue at hand.

**Comment by rsc** [comment]
41b3e8b said "**may** fix #353", not "fixes #353". Thanks GitHub! Reopening.
Anyway, if it is fixed, feel free to close. :-)

**Comment by mk-f** [comment]
It is, thank you!
